### PR TITLE
Github notification handling

### PR DESCRIPTION
**Pull request by maallen**
Adds support to notify Github for branch & string checker notifications.

Adds ability to set a Github commit status for string checker results from the cli jar.

### CODE REVIEW

File: cli/src/main/java/com/box/l10n/mojito/cli/command/extractioncheck/ExtractionCheckNotificationSenderGithub.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import com.box.l10n.mojito.github.GithubClients;
+import com.box.l10n.mojito.thirdpartynotification.github.GithubIcon;
+import com.google.common.base.Strings;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.kohsuke.github.GHCommitState;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+
+@Configurable
```
Comment (aurambaj) [inline cli/src/main/java/com/box/l10n/mojito/cli/command/extractioncheck/ExtractionCheckNotificationSenderGithub.java:13]: brushed on a new pattern that avoids using Aspectj for DI and pass dependencies by ctor instead. Maybe worth a refactoring this older code at some point. 

that code initialize either by CLI args or fully from springboot configuration

https://github.com/box/mojito/blob/f6f74037b984110563db7ea0ae58369cfa029278/cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionDiffNotificationCommand.java#L228

**Review by aurambaj — APPROVED** [review]

File: webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java [hunk 2]
```diff
@@ -17,40 +20,100 @@ public class BranchNotificationMessageSenderGithub implements BranchNotification
...
+    } catch (GithubException e) {
+      throw new BranchNotificationMessageSenderException(e);
+    }
   }
 
   @Override
   public void sendTranslatedMessage(String branchName, String username, String messageId)
       throws BranchNotificationMessageSenderException {
-    logger.info("sendTranslatedMessage");
+    logger.debug("sendTranslatedMessage to: {}", "https://github.com/" + branchName);
```
[the remark this answers is not in the document]
Comment (maallen) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: Are we not tied to `github.com` as we're using the GH API with a GH app to send these notifications?

[the remark this answers is not in the document]
Comment (ehoogerbeets) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: github enterprise -> many clients use mojito with internal github servers. Same API, different server URI

[the remark this answers is not in the document]
Comment (maallen) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: Will make this value configurable in that case

[the remark this answers is not in the document]
Comment (aurambaj) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: granted this is just a log, and doesn't change whatever is happening under the hood in the `githubClient`

[the remark this answers is not in the document]
Comment (ehoogerbeets) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: maybe there are other places in the code hard-coded to github.com?

[the remark this answers is not in the document]
Comment (aurambaj) [reply webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/github/BranchNotificationMessageSenderGithub.java:73]: I think we're ok. If we eventually need on prem, we just need to pass the configuration down to the client with `GitHubBuilder#withEndpoint`

**Pull request merged**